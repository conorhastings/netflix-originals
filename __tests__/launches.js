import React from 'react';
import renderer from 'react-test-renderer';
import Launches, { Launch, ShowAll, AllLaunches } from '../src/components/launches.js';


test('Launches component renders without error', () => {
  const tree = renderer.create(
  	<Launches 
  	  launches={[{ id: 1, title: 'coool' }]}
  	  dayHeight={77}
      day={1}
  	/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Launch component renders without error', () => {
  const tree = renderer.create(
  	<Launch title="coool" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ShowAll component renders without error', () => {
  const tree = renderer.create(
  	<ShowAll remainingMovies={3} onClick={() => {}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('AllLaunches component renders without error', () => {
  const tree = renderer.create(
    <AllLaunches launches={[{ id: 1, title: 'coool' }]} day={1} onClickX={() => {}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});