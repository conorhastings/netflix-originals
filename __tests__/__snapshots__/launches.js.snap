// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllLaunches component renders without error 1`] = `
<div
  className="css-12efcmn"
>
  <div
    className="css-p6nzn0"
  >
    <div
      className="css-gg4vpm"
    >
      <div>
        1
      </div>
      <div
        className="css-4g6ai3"
        onClick={[Function]}
      >
        Ã—
      </div>
    </div>
    <div
      className="css-111ae3z"
    >
      <div
        className="css-18n2zzf"
      >
        coool
      </div>
    </div>
  </div>
</div>
`;

exports[`Launch component renders without error 1`] = `
<div
  className="css-18n2zzf"
>
  coool
</div>
`;

exports[`Launches component renders without error 1`] = `
<div>
  <div
    className="css-18n2zzf"
  >
    coool
  </div>
</div>
`;

exports[`ShowAll component renders without error 1`] = `
<div
  className="css-1ngsyok"
  onClick={[Function]}
>
  Show All (
  3
   remaining)
</div>
`;
